src/main/java/teammates/common/datatransfer/questions/FeedbackMsqQuestionDetails.java

```diff
@@ -134,6 +163,28 @@ public boolean isChangesRequiresResponseDeletion(FeedbackQuestionDetails newDeta
             return true;
         }
 
+        if (this.maxSelectableChoices == Integer.MIN_VALUE && newMsqDetails.maxSelectableChoices != Integer.MIN_VALUE) {
+            // Delete responses if max selectable restriction is newly added
+            return true;
+        }
+
+        if (this.minSelectableChoices == Integer.MIN_VALUE && newMsqDetails.minSelectableChoices != Integer.MIN_VALUE) {
+            // Delete responses if min selectable restriction is newly added
+            return true;
+        }
+
+        if (this.minSelectableChoices != Integer.MIN_VALUE && newMsqDetails.minSelectableChoices != Integer.MIN_VALUE
+                && this.minSelectableChoices > newMsqDetails.minSelectableChoices) {
```

Comment:
I may understand this wrongly, but shouldn't it be the other way around? If the new details enforces 2 and the old one enforces 3, then any old answer is as valid for the new details. But if the new details enforces 3 and the old one enforces 2, some answers which pick only 2 options will be invalid.